### PR TITLE
Slice library cleanup, remove dead code and other minor fixes

### PR DESCRIPTION
**Pull request by pepone**
A follow-up PR to #1674, removes unused code from Slice library

### CODE REVIEW

File: cpp/src/Slice/Parser.cpp [hunk 1]
```diff
@@ -1562,28 +1525,20 @@ Slice::Container::createDictionary(const string& name, const TypePtr& keyType, c
     ContainedList matches = _unit->findContents(thisScope() + name);
     if(!matches.empty())
     {
-        DictionaryPtr p = dynamic_pointer_cast<Dictionary>(matches.front());
-        if(p)
-        {
-            if(_unit->ignRedefs())
```
Comment (pepone) [inline cpp/src/Slice/Parser.cpp:1568]: ignRedefs was always false, I think it was used before by local Slice and scripting languages.

File: cpp/src/Slice/Parser.cpp [hunk 2]
```diff
@@ -5163,11 +4747,6 @@ Slice::Const::Const(const ContainerPtr& container, const string& name, const Typ
     _value(value),
     _literal(literal)
 {
-    if(valueType == 0)
```
Comment (pepone) [inline cpp/src/Slice/Parser.cpp:5166]: this seems like some leftover from debugging the compilers

File: cpp/src/Slice/Parser.cpp [hunk 3]
```diff
@@ -5133,13 +4724,6 @@ Slice::Const::containedType() const
     return ContainedTypeConstant;
 }
 
-bool
-Slice::Const::uses(const ContainedPtr& contained) const
-{
-    ContainedPtr contained2 = dynamic_pointer_cast<Contained>(_type);
-    return (contained2 && contained2 == contained);
-}
```
Comment (pepone) [inline cpp/src/Slice/Parser.cpp:5141]: this code was wrong with the upgrade to shared_ptr, and still everything worked! noticed it is not used anymore.

**Review by externl — COMMENTED** [review]
Looks good

**Pull request merged**